src/transformers/pipelines.py

```diff
@@ -1435,17 +1457,31 @@ def group_sub_entities(self, entities: List[dict]) -> dict:
             entities (:obj:`dict`): The entities predicted by the pipeline.
         """
         # Get the first entity in the entity group
-        entity = entities[0]["entity"]
-        scores = np.mean([entity["score"] for entity in entities])
+        entity = entities[0]["entity"].split("-")[-1]
+        scores = np.nanmean([entity["score"] for entity in entities])
         tokens = [entity["word"] for entity in entities]
 
         entity_group = {
             "entity_group": entity,
             "score": np.mean(scores),
-            "word": self.tokenizer.convert_tokens_to_string(tokens),
+            "word": self.convert_tokens_to_string(tokens),
         }
         return entity_group
 
+    def is_subword_fn(self, token: str) -> bool:
+        if token.startswith("##"):
+            return True
+        return False
```

Comment:
This is very specific to BERT

Reply:
I provide alternatives for those BERT specific functions; the user can override the function by providing their own tokenizer.is_subword_fn and tokenizer.convert_tokens_to_string. while it is not a clean solution it is flexible. I would like it if there was a property like tokenizer.subword_prefix for all tokenizers, then I could use that. (I remember I saw something similar somewhere maybe in the fast tokenizers?)

Reply:
I'm pretty sure for slow tokenizers one can use `tokenizer.convert_tokens_to_string`, while for fast tokenizer you can use `token_to_word` to obtain the indices and construct words.

I don't think expecting the user to provide a method to join tokens is the best we can do here. Pipelines are meant as a way for users to use the models without needing to understand what happens under the hood, having them specify a tokenizer-specific method to join tokens goes against that goal.

Reply:
I agree about the pipelines, hadn't liked my previous hacky solution myself either. I have changed tokenizer.convert_tokens_to_string bit as suggested https://github.com/huggingface/transformers/pull/5970#discussion_r511790482. and also the subword detection logic.https://github.com/huggingface/transformers/pull/5970#discussion_r511789502